src/libs/actions/Report.ts

```diff
@@ -2668,9 +2668,15 @@ function openReportFromDeepLink(url: string) {
     // Navigate to the report after sign-in/sign-up.
     InteractionManager.runAfterInteractions(() => {
         Session.waitForUserSignIn().then(() => {
-            Onyx.connect({
+            const connection = Onyx.connect({
                 key: ONYXKEYS.NVP_ONBOARDING,
                 callback: (onboarding) => {
+                    if (onboarding) {
+                        // Once the onboarding data is available, we want to disconnect the connection
+                        // so it won't trigger the deeplink again every time the data is changed, for example, when relogin.
+                        Onyx.disconnect(connection);
+                    }
```

Comment:
We're disconnecting onyx before the onboarding is complete. This caused https://github.com/Expensify/App/issues/48475.